### ISSUE TITLE
Change '/give' to '/minecraft:give'

### ISSUE THREAD

**Issue by Flexo013** [Flexo013/HeadsWebsite#16]
On vanilla servers our generated commands work fine. However when other services try to take over `/give` the command might not work. Both Essentials and Bukkit have their own `/give` which does not allow for our base64 hack. To fix this we can replace `/give` with `/minecraft:give`. This way we make sure that use Minecraft's `/give`-command and not any other.

_(Made this ticket since a friend with a Bukkit 1.7.9 server reported that the commands were not working. With this suggested enhancement it did work.)_

**Comment by Fancypants** [comment]
Excellent find, Mark and I were already expecting this problem and replacing the /give with /minecraft:give seems like a good solution. I'll probably need to write a small script to change all database values. Did you test Essentials and Bukkit with the /minecraft:give command yourself? Will need someone to verify this before changing all commands, essentials seems to be very aggressive in taking over commands.

**Comment by Flexo013** [comment]
Have confirmed that this works on Bukkit with Essentials.

**Comment by Fancypants** [comment]
We tried our commands on the new spigot (bukkit) 1.8 builds, it indeed did not work. However it will work with essentials! Further investigations showed us the command /give will work for everything else except for skulls (where the "minecraft:" prefix is needed somewhere in the command, either as /minecraft:give or minecraft:skull). I believe this is an inconsistency in spigot since other items will work without the prefix. Apparently the spigot give command does not know how to use our nbt tag.

In short our commands will work on:
- Vanilla minecraft 1.8
- Spigot 1.8 with essentials

Will not work on:
- Spigot 1.8

Possible solutions:
1. Request spigot to look into this inconsistency and fix the /give command for all items
2. Incorporate /minecraft:give with our /give command.

I don't like fixing stuff that should be fixed from the source, will try 1 first, if spigot doesn't fix it I will create a script for solution 2. To be continued...

**Comment by Fancypants** [comment]
Seems like bukkit/spigot does not implement all vanilla minecraft functionality, sadly they do mess up commands like ours by only implementing half. 

Someone told me we can let spigot override commands in commands.yml:

```
command-block-overrides:
- '*'
```

If this works it means our commands work in bukkit/spigot with(out) essentials and in vanilla minecraft.

**Comment by Flexo013** [comment]
That is nice, but our users would have to configure that manually. Maybe they might be op but not have access to the files to change the settings.

We don't really lose anything by extending the commands a bit. The extra length on already 200 characters is not really a significant change. I also see no reason why the extra length would be a problem.
By replacing this we simply improve our functionality at the loss of nothing.

As far as the script to fix this I suggest using RegEx to just check for /give and replace it with /minecraft:give.

**Comment by Flexo013** [comment]
Closed since the commands would no longer work for Vanilla.
